pkgs/merkle-tree-wasm/src/merkle_tree_wasm.rs

```diff
@@ -127,11 +125,12 @@ impl<F: PrimeField, const WIDTH: usize> MerkleTree<F, WIDTH> {
 
         let mut current_layer = &self.layers[0];
 
-        let mut leaf_index = self
-            .leaves
-            .iter()
-            .position(|&x| x == leaf)
-            .expect("MerkleProof: Leaf is not found");
+        let leaf_index_result = self.leaves.iter().position(|&x| x == leaf);
+
+        let mut leaf_index = match leaf_index_result {
+            Some(index) => index,
+            None => return Err("MerkleProof: Leaf is not found, please consider using an address belongs to the anonymous set.".to_string()),
```

Comment:
Is it possible to have access to the non hashed leaf here? I mean the address we want to prove is part of the tree?
I'd like to have a message like.

`Merkle proof error: leaf not found. Address {address} is not part of the addresses set`.

It is clearer and more user friendly.
But I'd like to print out the `0x` address in the error message not the poseidon hash

Reply:
Well I dont see a reason for not doing this. 
Let me push that change.